tests/subsys/debug/gdbstub/pytest/test_gdbstub.py

```diff
@@ -4,31 +4,67 @@
 
 import os
 import subprocess
-from twister_harness import DeviceAdapter
 import sys
 import logging
 import shlex
+import re
+import pytest
+from twister_harness import DeviceAdapter
 
 ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
 sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts", "pylib", "twister"))
 from twisterlib.cmakecache import CMakeCache
 
 logger = logging.getLogger(__name__)
 
-
-def test_gdbstub(dut: DeviceAdapter):
-    """
-    Test gdbstub feature using a gdb script. We connect to the DUT and run some
-    basic gdb commands and evaluate return code to determine pass or failure.
-    """
+@pytest.fixture()
+def gdb_process(dut: DeviceAdapter, gdb_script, gdb_timeout):
     build_dir = dut.device_config.build_dir
-    cmake_cache = CMakeCache.from_file(build_dir / 'CMakeCache.txt')
-    gdb = cmake_cache.get('CMAKE_GDB', None)
-    assert gdb
+    cmake_cache = CMakeCache.from_file(os.path.join(build_dir, 'CMakeCache.txt'))
+    gdb_exec = cmake_cache.get('CMAKE_GDB', None)
+    assert gdb_exec
     source_dir = cmake_cache.get('APPLICATION_SOURCE_DIR', None)
     assert source_dir
-    cmd = [gdb, '-x', f'{source_dir}/run.gdbinit', f'{build_dir}/zephyr/zephyr.elf']
-    logger.info(f'Test command: {shlex.join(cmd)}')
-    result = subprocess.run(cmd, capture_output=True, text=True, timeout=20)
-    logger.debug('Output:\n%s' % result.stdout)
-    assert result.returncode == 0
+    build_image = cmake_cache.get('BYPRODUCT_KERNEL_ELF_NAME', None)
+    assert build_image
+    gdb_log_file = os.path.join(build_dir, 'gdb.log')
+    cmd = [gdb_exec, '-batch', '-ex', f'set logging file {gdb_log_file}',
+           '-x', f'{source_dir}/{gdb_script}', build_image]
+    logger.info(f'Run GDB: {shlex.join(cmd)}')
+    result = subprocess.run(cmd, capture_output=True, text=True, timeout=gdb_timeout)
+    logger.info(f'GDB ends rc={result.returncode}')
+    return result
+#
+
+@pytest.fixture(scope="module")
+def expected_app():
```

Comment:
> this is all related to one test, why do we need a fixture? They can be defined in the test_ itself.

just to follow the pytest way of 'fixturizing' test data from test logic.

Reply:
It is always good to have separated data from the logic, so the idea is good. Here it is only one test, so you can use just a method instead of fixture (however fixture is also ok)